subsys/mpsl/init/mpsl_init.c

```diff
@@ -52,25 +52,14 @@ static uint8_t __aligned(4) timeslot_context[TIMESLOT_MEM_SIZE];
 
 static void mpsl_low_prio_irq_handler(void)
 {
-	k_sem_give(&sem_signal);
+	k_work_submit_to_queue(&mpsl_work_q, &mpsl_low_prio_work);
 }
 
-static void signal_thread(void *p1, void *p2, void *p3)
+static void mpsl_low_prio_work_handler(struct k_work *item)
 {
-	ARG_UNUSED(p1);
-	ARG_UNUSED(p2);
-	ARG_UNUSED(p3);
+	ARG_UNUSED(item);
 
-	int errcode;
-
-	while (true) {
-		k_sem_take(&sem_signal, K_FOREVER);
-
-		errcode = MULTITHREADING_LOCK_ACQUIRE();
```

Comment:
why don't we need the multithreading locks anymore ? It seems the same locks are used by the hci driver for SDC calls. Were they used here only to prevent mpsl/mpsl threading issues or mpsl/sdc issues ?

Reply:
It's more than a year ago that I did this. I don't actually remember now.
@rugeGerritsen Do you?

Reply:
This looks wrong to me. We need to ensure all APIs are executed with a lock. I tried finding a case where adding the lock back could result in a deadlock, but I can't come up with a case. Can you try adding it back?

Reply:
Added back now. Could have been a change that was temporary needed when the multithreading lock was a semaphore instead of a mutex.